### ISSUE TITLE
Cannot delete services using label selector in 1.1.3

### ISSUE THREAD

**Issue by krmayankk** [kubernetes/kubernetes#22771]
$ cluster/kubectl.sh version
Client Version: version.Info{Major:"1", Minor:"1", GitVersion:"v1.1.3", GitCommit:"6a81b50c7e97bbe0ade075de55ab4fa34f049dc2", GitTreeState:"clean"}
Server Version: version.Info{Major:"1", Minor:"1", GitVersion:"v1.1.3", GitCommit:"6a81b50c7e97bbe0ade075de55ab4fa34f049dc2", GitTreeState:"clean"}
$ cluster/kubectl.sh get services
NAME                    CLUSTER_IP    EXTERNAL_IP      PORT(S)   SELECTOR                                     AGE
kubernetes              10.0.0.1      <none>           443/TCP   <none>                                       49d
shared-rptcoll1-0-phx   10.0.219.61   130.211.121.65   80/TCP    app-test=report-collector-report-collector   3m

$ cluster/kubectl.sh delete services -lapp-test
No resources found

**Comment by zhouhaibing089** [comment]
how about `kubectl delete services -l app-test=report-collector-report-collector`?

**Comment by krmayankk** [comment]
same issue
./cluster/kubectl.sh delete services -l app-test=report-collector-report-collector
No resources found

**Comment by zhouhaibing089** [comment]
`kubectl get services` shows what?

**Comment by AdoHe** [comment]
@krmayankk @zhouhaibing089 well I am afraid you misused this kubectl delete.

**Comment by AdoHe** [comment]
`kubectl delete services -l <label_name>=<label_value>`

**Comment by zhouhaibing089** [comment]
@AdoHe thanks, I mean the same ^_^.

**Comment by zhouhaibing089** [comment]
@AdoHe 
> how about kubectl delete services -l app-test=report-collector-report-collector?

I mean the same where `app-teset` is the label name and `report-collector-report-collector` is the label value.

**Comment by krmayankk** [comment]
pods,services works but just specifying the services doesnt work. I am not sure what is the thinking behind that implementation ?

**Comment by zhouhaibing089** [comment]
@krmayankk thanks for clarify this.

**Comment by AdoHe** [comment]
@krmayankk @zhouhaibing089 I did test on HEAD:
```
[tony@localhost kubernetes]$ cluster/kubectl.sh get services -o wide --show-labels=true
NAME         CLUSTER-IP   EXTERNAL-IP   PORT(S)   AGE       SELECTOR    LABELS
kubernetes   10.0.0.1     <none>        443/TCP   2m        <none>      component=apiserver,provider=kubernetes
my-service   10.0.0.254   <none>        80/TCP    1m        app=MyApp   name=testservice
[tony@localhost kubernetes]$ cluster/kubectl.sh delete services -l name=testservice
service "my-service" deleted
[tony@localhost kubernetes]$ cluster/kubectl.sh get services -o wide --show-labels=true
NAME         CLUSTER-IP   EXTERNAL-IP   PORT(S)   AGE       SELECTOR   LABELS
kubernetes   10.0.0.1     <none>        443/TCP   2m        <none>     component=apiserver,provider=kubernetes
```
would you mind giving a try on master branch?

**Comment by lavalamp** [comment]
Thanks for the report. I don't expect we'll backport a fix into 1.1. Please reopen if that's super important for your use case.
